### ISSUE TITLE
The shelf has too much space on top (iPhone)

### ISSUE THREAD

**Issue by MaverickWar** [Simbul/baker#1023]
Hi guys, can anyone tell me why there is such a massive gap at the top of baker shelf, can this be removed as there is a lot of wasted real estate there particularly when it takes up a third of an iPhone screen.
![image](https://f.cloud.github.com/assets/4552051/810927/e2196a0c-eec4-11e2-9fe7-6a7dc23fa43a.jpg)

**Comment by folletto** [comment]
The "massive gap" is the space defined to add you magazine details, logo, and creativity. It's clear from the screenshot you're not using that... but that's how Baker is designed currently. If you check the existing Baker png files and how other magazines manage it, you see that they all have that space used for the actual magazine header. :)

**Comment by MaverickWar** [comment]
That's great but as I'm not using it, is there anything I can edit to remove that space while keeping things functioning?

**Comment by folletto** [comment]
You need to change the code yourself, we don't provide a way to change it, I'm sorry.

**Comment by MaverickWar** [comment]
Could you at least point me in the right direction with a file I need to edit? I must say that given this is an open source project, I would have thought you could offer a little help and guidance on these matters without it impacting your time to much... After all, I'm not asking for a rewrite, just help to make a small customisation.

**Comment by folletto** [comment]
I don't know if it's a small customization, do you? :)
It might be as simple as changing a line, or as complex as rewriting the entire rendering code of the shelf, I frankly don't know, you need to look into it.

Please however don't ask people that is already helping and often answering within hours (or minutes) to do more. We are here to help each other, and sometimes there are people that are helping a huge lot. Open source means that we support each other in a community, not that the core team must answer everyone. ;)

—

That said, sometimes is just enough to ask. Try looking around the `BakerShelf/ShelfViewController.m` file, around line `267` you should be able to see:

```
self.gridView.frame = CGRectMake(0, bannerHeight, width, height - bannerHeight);
```

That clearly points out to the `bannerHeight`, declared on line `262`:

```
int bannerHeight = [ShelfViewController getBannerHeight];
```

Pointing again to the `getBannerHeight:` function, on line `653`.

Try playing around with these and the `self.gridView`... :)

**Comment by MaverickWar** [comment]
This was exactly what I was looking for, thank you kindly sir! Fits much better now and on small iPhone screens, I have larger screen real estate now

![image](https://f.cloud.github.com/assets/4552051/816188/31cf77fe-ef3c-11e2-8488-9efd81ccc536.jpg)

**Comment by fturner19** [comment]
@MaverickWar - That's looking really good man! Have you tried the new pull with the info screen added to the left? Would really like to see what it would look like on the iphone. But good work on your overall design.

**Comment by MaverickWar** [comment]
Thanks man! I tried adding the code for the info thing as per instructions given on an external site, but I broke my build somehow, whenever I mess around with code I seem to break something, even changing this seems to have done something.......

Whenever I remove an issue from magrocket server and refresh in the app while its open it loads the deleted magazine at the bottom. It only removes the magazine when I close app entirely, before this edit, if I removed an issue and refreshed while app was open, the issue would just disappear.... So not sure what I've done.

I would like to add the info screen though for sure.
